python/activator/activator.py

```diff
@@ -84,12 +84,15 @@
 
 storage_client = boto3.client('s3', endpoint_url=s3_endpoint)
 
+central_butler = get_central_butler(calib_repo, instrument_name)
+# local_repo is a temporary directory with the same lifetime as this process.
+local_repo = make_local_repo(local_repos, central_butler, instrument_name)
```

Comment:
I can image why we can't use a context manager for cleanup here; do we need to worry about that, or is the lifetime of this process tied to some kind of more complete reset of the compute resource that will take care of that?

Reply:
For performance reasons, we want to reuse the same local repo for processing multiple exposures. And AFAIK Flask doesn't let us pass in arguments to `next_visit_handler`, so it needs to be (directly or indirectly) a global. I think that precludes use of a context manager.

Basically, the way the Prompt Processing architecture is set up, the Python process represented by `activator.py` has the same lifetime as the (virtual) filesystem it's running on. I don't know if that answers your second question.